dfastmi/io/RiverParameterData.py

```diff
@@ -0,0 +1,674 @@
+                            )
+                        )
+                values_per_branch.append((vals[0], vals[1], vals[2], vals[3]))
+            all_values.append(values_per_branch)
+        return all_values
+
+    def collect_valuesN(
+        self,
+        key: str
+    ) -> List[List[Tuple[float, ...]]]:
+        """
+        This routines collects entries of type Tuple[float, ...]
+
+        Arguments
+        ---------
+        key : str
+            The name of the parameter for which the values are to be retrieved.
+
+        Returns
+        -------
+        data : List[List[Tuple[float, ...]]]
+            A list of lists. Each list contains per reach within the corresponding
+            branch a list of floats.
+        """
+        vals: Tuple[float, ...]
+        self.is_initialized()        
+        try:
+            g_val = self._config["General"][key]
+        except:
+            g_val = ""
+        all_values = []
+        for ib in range(len(self._branches)):
+            branch = self._branches[ib]
+            try:
+                b_val = self._config[branch][key]
+            except:
+                b_val = g_val
+            values_per_branch = []
+            for i in range(self._nreaches[ib]):
+                stri = str(i + 1)
+                try:
+                    val = self._config[branch][key + stri]
+                except:
+                    val = b_val
+                vals = tuple(float(x) for x in val.split())
+                values_per_branch.append(vals)
+            all_values.append(values_per_branch)
+        return all_values
+    
+    def config_get_bool(
+        self,
+        group: str,
+        key: str,
+        default: Optional[bool] = None,
+    ) -> bool:
+        """
+        Get a boolean from a selected group and keyword in the analysis settings.
+
+        Arguments
+        ---------
+        group : str
+            Name of the group from which to read.
+        key : str
+            Name of the keyword from which to read.
+        default : Optional[bool]
+            Optional default value.
+
+        Raises
+        ------
+        Exception
+            If the keyword isn't specified and no default value is given.
+
+        Returns
+        -------
+        val : bool
+            Boolean value.
+        """
+        try:
+            str = self._config[group][key].lower()
+            val = (
+                (str == "yes")
+                or (str == "y")
+                or (str == "true")
+                or (str == "t")
+                or (str == "1")
+            )
+        except:
+            if not default is None:
+                val = default
+            else:
+                raise Exception(
+                    'No boolean value specified for required keyword "{}" in block "{}".'.format(
+                        key, group
+                    )
+                )
+        return val
+
+    def config_get_int(
```

Comment:
is this the correct method name? config makes it sound like it belongs in the Config class.